include/zephyr/device.h

```diff
@@ -906,9 +906,24 @@ static inline bool z_impl_device_is_ready(const struct device *dev)
 		.state = (state_),                                             \
 		.data = (data_),                                               \
 		IF_ENABLED(CONFIG_DEVICE_DEPS, (.deps = (deps_),)) /**/        \
-		IF_ENABLED(CONFIG_PM_DEVICE, ({ .pm_base = (pm_),)}) /**/         \
+			Z_DEVICE_INIT_PM_BASE(pm_)                             \
 	}
 
+/*
+ * Anonymous unions require C11. Some pre-C11 gcc versions have early
+ * support for anonymous unions but they require these braces when
+ * combined with C99 designated initializers. These braces are
+ * compatible with any C version but not with C++20.  For more obscure
+ * C/C++ compatibility details see commit c15f029a7108 and PR #69411.
```

Comment:
please avoid referring to PRs in documentation

Reply:
I usually do but this issue is really complex and that PR discussion is quite important.

Reply:
PRs content is not part of Zephyr git history. Who knows if Github will stay Zephyr's VCS provider forever.

Reply:
I'm with Gerard on this, just refer to the commit, one can find back the PR out of band but really the commit should have enough context regardless

If you really want to link a PR maybe you can do it on this comment using the `Link:` tag and the full URL?

Reply:
> Who knows if Github will stay Zephyr's VCS provider forever.

I'm aware and I always try to avoid references to PRs when the git history is enough. But I think there can be some exceptions. It's not because something may disappear one day that it should not be referenced. If Zephyr migrates away from Github one day, then it should definitely export all past review discussions and migrate or archive them somewhere and then PR numbers would still be useful. I've been performing such archeology in the past and any reference helps.

Of course documentation is even better but (1) it's not mutually exclusive (2) not obvious where in the docs this obscure issue belongs (3) more "overhead" :-)

> PRs content is not part of Zephyr git history.

True but the git history does not exist in a complete vacuum either.

> If you really want to link a PR maybe you can do it on this comment using the Link: tag and the full URL?

That would work for me.

Or I can just delete the PR number entirely if it's really import for you. I think that would be a loss but it's not that big of a deal for me.

Reply:
Full link of external references in the commit I think are fine, Linux does it as well, it's more my opinion than a general rule but I think it's a good idea and I occasionally do it myself too.